subcommand.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+)
+
+type SubCommand struct {
+	cmd       *exec.Cmd
+	PipeRead  *os.File
+	pipeWrite *os.File
+	Err       error
+}
+
+func NewSubCommand(args []string) SubCommand {
+	pipeRead, pipeWrite, err := os.Pipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	env := append(os.Environ(), "GODEBUG=gctrace=1")
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Env = env
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = pipeWrite
+
+	return SubCommand{
+		cmd:       cmd,
+		PipeRead:  pipeRead,
+		pipeWrite: pipeWrite,
+	}
+}
+
+func (s *SubCommand) Run() {
+	if err := s.cmd.Run(); err != nil {
```

Comment:
`s.Err` is checked in the main function. However, I don't need a mutex.

There is two ways at the moment to know if `s.Run()` returned an error:

1. Call `Run()` synchronously and check `s.Err`.
2. Use a goroutine to call `Run()` and read the pipe until you reach EOF, and then check `s.Err`.

In the main function, I am in the second scenario as I am waiting for the parser to send `true` through `parser.done` (see https://github.com/davecheney/gcvis/pull/19/files#diff-7ddfb3e035b42cd70649cc33393fe32cR55 and https://github.com/davecheney/gcvis/pull/19/files#diff-f615844d3497ff38db57e459d6ef657bR78). Receiving `parser.done` means that `sc.Scan` reached EOF.